lib/openhab/dsl/things.rb

```diff
@@ -86,7 +86,7 @@ class Things
         # Gets a specific thing by name in the format binding_id:type_id:thing_id
         # @return Thing specified by name or nil if name does not exist in thing registry
         def [](uid)
-          thing_uid = org.openhab.core.thing.ThingUID.new(*uid.split(':'))
+          thing_uid = uid.is_a?(String) ? org.openhab.core.thing.ThingUID.new(*uid.split(':')) : uid
```

Comment:
Maybe switch this. If it is a UID, do nothing. Otherwise, to_s and split it. Thinking of StringState, which isn't actually a string, but is implicitly convertible to one

Reply:
+1